src/Sarif.Viewer.VisualStudio/Controls/InfoBar.cs

```diff
@@ -197,5 +217,21 @@ public void OnActionItemClicked(IVsInfoBarUIElement infoBarUIElement, IVsInfoBar
                 this.clickAction?.Invoke(actionItem);
             }
         }
+
+        private static async Task AddInfoBarIfRequiredAsync(
+            ExceptionalConditions detectedConditions,
+            ExceptionalConditions individualCondition,
+            string message,
+            ImageMoniker imageMoniker)
+        {
+            // It's safe to access and update the dictionary because this method is called inside
+            // the write lock.
+            if ((detectedConditions & individualCondition) == individualCondition && !s_infoBarToConditionDictionary.Values.Contains(individualCondition))
```

Comment:
I have a feeling that the ExceptionalConditions value should be unique in this Dictionary. Don't you think the key should be an ExceptionalConditions value, instead of a InforBar object?
Feel free to correct me if I am wrong :)

Reply:
With ExceptionalConditions value as key, if an InfoBar already exists, updating it's message, and reloading/refreshing the infoBar display probably is a way for multiple similar categories of information to display.

Reply:
As the code stands, the logic enforces that there's only one `InfoBar` per condition value. I actually did it that way for a reason, but since it's not clear the the reader I'll add a "why" comment explaining why.

---
In reply to: [527201705](https://github.com/microsoft/sarif-visualstudio-extension/pull/278#discussion_r527201705) [](ancestors = 527201705)

Reply:
You're right, it does seem natural to use the condition as the key. I'll add a comment explaining why I did it this way.

And is so often the case when you write a comment, I might discover that there's a better way :-)

---
In reply to: [527210678](https://github.com/microsoft/sarif-visualstudio-extension/pull/278#discussion_r527210678) [](ancestors = 527210678)

Reply:
> Because `detectedConditions` is a `Flags` enum -- the same log can exhibit multiple exceptional conditions. I need to know if this particular bit is turned on, regardless of what other bits might be turned one.
> 
> In reply to: [527204867](https://github.com/microsoft/sarif-visualstudio-extension/pull/278#discussion_r527204867) [](ancestors = 527204867)

Get it. Didn't notice this. Thanks for the info.

Reply:
Ok, I tried coding it the other way and it actually makes the code a bit more complicated. So I've done the due diligence and I'm now just going to add the comment that explains why it is as it is.

---
In reply to: [527229642](https://github.com/microsoft/sarif-visualstudio-extension/pull/278#discussion_r527229642) [](ancestors = 527229642,527210678)

Reply:
Done.

---
In reply to: [527281811](https://github.com/microsoft/sarif-visualstudio-extension/pull/278#discussion_r527281811) [](ancestors = 527281811,527229642,527210678)